sdk/tables/Azure.Data.Tables/tests/TableClientTests.cs

```diff
@@ -0,0 +1,130 @@
+
+            Assert.That(async () => await client_Instrumented.UpsertAsync(entityWithoutPK), Throws.InstanceOf<ArgumentException>(), $"The method should validate the entity has a {TableConstants.PropertyNames.PartitionKey}.");
+
+            Assert.That(async () => await client_Instrumented.UpsertAsync(entityWithoutRK), Throws.InstanceOf<ArgumentException>(), $"The method should validate the entity has a {TableConstants.PropertyNames.RowKey}.");
+
+            Assert.That(async () => await client_Instrumented.UpdateAsync(null, "etag"), Throws.InstanceOf<ArgumentNullException>(), "The method should validate the entity is not null.");
+
+            Assert.That(async () => await client_Instrumented.UpdateAsync(validEntity, null), Throws.InstanceOf<ArgumentNullException>(), "The method should validate the entity is not null.");
+
+            Assert.That(async () => await client_Instrumented.UpdateAsync(entityWithoutPK, "etag"), Throws.InstanceOf<ArgumentException>(), $"The method should validate the entity has a {TableConstants.PropertyNames.PartitionKey}.");
+
+            Assert.That(async () => await client_Instrumented.UpdateAsync(entityWithoutRK, "etag"), Throws.InstanceOf<ArgumentException>(), $"The method should validate the entity has a {TableConstants.PropertyNames.RowKey}.");
+
+            Assert.That(async () => await client_Instrumented.MergeAsync(null, "etag"), Throws.InstanceOf<ArgumentNullException>(), "The method should validate the entity is not null.");
+
+            Assert.That(async () => await client_Instrumented.MergeAsync(entityWithoutPK, "etag"), Throws.InstanceOf<ArgumentException>(), $"The method should validate the entity has a {TableConstants.PropertyNames.PartitionKey}.");
+
+            Assert.That(async () => await client_Instrumented.MergeAsync(entityWithoutRK, "etag"), Throws.InstanceOf<ArgumentException>(), $"The method should validate the entity has a {TableConstants.PropertyNames.RowKey}.");
+        }
+
+        [Test]
+        public void GetSasBuilderPopulatesPermissionsAndExpiry()
+        {
+            var expiry = DateTimeOffset.Now.AddDays(1);
+            var permissions = TableSasPermissions.All;
+
+            var sas = client_Instrumented.GetSasBuilder(permissions, expiry);
+
+            Assert.That(sas.Permissions, Is.EqualTo(permissions.ToPermissionsString()));
+            Assert.That(sas.ExpiresOn, Is.EqualTo(expiry));
+        }
+
+        [Test]
+        public void GetSasBuilderPopulatesRawPermissionsAndExpiry()
+        {
+            var expiry = DateTimeOffset.Now.AddDays(1);
+            var permissions = TableSasPermissions.All;
+
+            var sas = client_Instrumented.GetSasBuilder(permissions.ToPermissionsString(), expiry);
+
+            Assert.That(sas.Permissions, Is.EqualTo(permissions.ToPermissionsString()));
+            Assert.That(sas.ExpiresOn, Is.EqualTo(expiry));
+        }
+
+        [Test]
+        public async Task CreateCallsInternalCreate()
```

Comment:
Is it worth having as a  unit test?

Reply:
Marginal, I was just covering the code path as no other test did